test/declaration_test.ts

```diff
@@ -256,6 +256,31 @@ class X {
   X.fakeConstructor$();
   external x();
 }`);
+    });
+    it('should emit extension methods to return Futures from methods that return Promises', () => {
+      expectTranslate(`declare interface MyMath {
+        randomInRange(start: number, end: number): Promise<number>;
+      }`).to.equal(`import "dart:async" show Completer;
+import "package:js/js_util.dart" show promiseToFuture;
+
+@anonymous
+@JS()
+abstract class MyMath {}
+
+@JS('MyMath')
+abstract class _MyMath {
+  Promise<num> randomInRange(num start, num end);
+}
+
+extension on MyMath {
+  Future randomInRange(num start, num end) {
+    final _MyMath t = this as _MyMath;
```

Comment:
remove the `as _MyMath` it shouldn't be needed and is hard for dart2js to optimize out.